diff -urN a/Basement/Types/OffsetSize.hs b/Basement/Types/OffsetSize.hs
--- a/Basement/Types/OffsetSize.hs	2019-06-30 10:22:18.000000000 +0300
+++ b/Basement/Types/OffsetSize.hs	2021-02-05 13:38:43.741587600 +0300
@@ -38,7 +38,6 @@
     , sizeOfE
     , csizeOfOffset
     , csizeOfSize
-    , sizeOfCSSize
     , sizeOfCSize
     , Countable
     , Offsetable
@@ -52,7 +51,6 @@
 import GHC.Word
 import GHC.Int
 import GHC.Prim
-import System.Posix.Types (CSsize (..))
 import Data.Bits
 import Basement.Compat.Base
 import Basement.Compat.C.Types
@@ -237,14 +235,6 @@
 csizeOfOffset (Offset (I# sz)) = CSize (W64# (int2Word# sz))
 #endif
 
-sizeOfCSSize :: CSsize -> CountOf Word8
-sizeOfCSSize (CSsize (-1))      = error "invalid size: CSSize is -1"
-#if WORD_SIZE_IN_BITS < 64
-sizeOfCSSize (CSsize (I32# sz)) = CountOf (I# sz)
-#else
-sizeOfCSSize (CSsize (I64# sz)) = CountOf (I# sz)
-#endif
-
 sizeOfCSize :: CSize -> CountOf Word8
 #if WORD_SIZE_IN_BITS < 64
 sizeOfCSize (CSize (W32# sz)) = CountOf (I# (word2Int# sz))
