diff -urN a/cbits/double.c b/cbits/double.c
--- a/cbits/double.c	2017-01-30 14:15:18.000000000 +0300
+++ b/cbits/double.c	2021-02-03 11:24:57.539272500 +0300
@@ -27,6 +27,7 @@
     return ux.w == uy.w;
 }
 
+#ifndef _MSC_VER
 double
 copysign (double x, double y)
 {
@@ -38,6 +39,7 @@
     uz.w = sign | val;
     return uz.d;
 }
+#endif
 
 /* ported from tango/math/IEEE.d nextup */
 double
diff -urN a/cbits/float.c b/cbits/float.c
--- a/cbits/float.c	2017-01-30 14:15:18.000000000 +0300
+++ b/cbits/float.c	2021-02-03 11:24:57.559481700 +0300
@@ -28,6 +28,7 @@
     return ux.w == uy.w;
 }
 
+#ifndef _MSC_VER
 float
 copysignf (float x, float y)
 {
@@ -39,6 +40,7 @@
     uz.w = sign | val;
     return uz.f;
 }
+#endif
 
 /* ported from tango/math/IEEE.d nextupf */
 float
