diff -urN persistent-sqlite-2.6.3/persistent-sqlite.cabal persistent-sqlite-2.6.3p/persistent-sqlite.cabal
--- persistent-sqlite-2.6.3/persistent-sqlite.cabal	2017-10-15 08:18:38.000000000 +0300
+++ persistent-sqlite-2.6.3p/persistent-sqlite.cabal	2017-11-28 17:01:05.657569700 +0300
@@ -44,6 +44,7 @@
     exposed-modules: Database.Sqlite
                      Database.Persist.Sqlite
     ghc-options:     -Wall
+    cc-options:      -std=c99
     if flag(systemlib)
         if flag(use-pkgconfig)
             pkgconfig-depends: sqlite3
@@ -52,11 +53,11 @@
     else
         c-sources:   cbits/sqlite3.c
         include-dirs: cbits
-        cc-options:  -fPIC -std=c99
 
     c-sources: cbits/config.c
 
     if !os(windows)
+        cc-options:  -fPIC
         extra-libraries: pthread
 
 source-repository head
