diff -urN unix-compat-0.5/cbits/mktemp.c unix-compat-0.5p/cbits/mktemp.c
--- unix-compat-0.5/cbits/mktemp.c	2017-11-07 02:19:40.000000000 +0300
+++ unix-compat-0.5p/cbits/mktemp.c	2017-11-28 10:48:54.817273000 +0300
@@ -33,7 +33,11 @@
  * SUCH DAMAGE.
  */
 
+#ifdef __MINGW32__
 #include <sys/param.h>
+#else
+#define MAXPATHLEN _MAX_PATH
+#endif
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <errno.h>
